inst/include/RcppEigenWrap.h

```diff
@@ -80,16 +80,19 @@ namespace Rcpp{
 
         // for plain dense objects
         template <typename T>
-        SEXP eigen_wrap_plain_dense( const T& obj, Rcpp::traits::true_type ){
-			typename Eigen::internal::conditional<T::IsRowMajor,
-												  Eigen::Matrix<typename T::Scalar,
-																T::RowsAtCompileTime,
-																T::ColsAtCompileTime>,
-												  const T&>::type objCopy(obj);
-            int m = obj.rows(), n = obj.cols();
-			R_xlen_t size = static_cast<R_xlen_t>(m) * n;
-			SEXP ans = PROTECT(::Rcpp::wrap(objCopy.data(), objCopy.data() + size));
-            if( T::ColsAtCompileTime != 1 ) {
+        SEXP eigen_wrap_plain_dense( const T& obj, Rcpp::traits::true_type ) {
+            typename Eigen::internal::conditional<
+                T::IsRowMajor,
+                Eigen::Matrix<typename T::Scalar,
+                              T::RowsAtCompileTime,
+                              T::ColsAtCompileTime>,
+                const T&>::type objCopy(obj);
+            R_xlen_t m = obj.rows(), n = obj.cols(), size = m * n;
+            SEXP ans = PROTECT(::Rcpp::wrap(objCopy.data(), objCopy.data() + size));
+            if ( T::ColsAtCompileTime != 1 ) {
+                if (m > INT_MAX || n > INT_MAX) {
```

Comment:
Shouldn't this be immediately after the dimension definition?

Reply:
Yes... I've just committed again to my fork to address that, but it won't appear here until the PR is reopened. (I think?)

Reply:
Tacking on: Is `objCopy` really a copy of `obj`? Why do we need `objCopy.data()` here, rather than `obj.data()`?

Reply:
Agreed that it may needs a cleanup as the `Rcpp:;stop()` also breaks the protect/unprotect dance.

The  `const T& obj` to `<.... const T&>::type objCopy(obj)` dance is likely due to some TMP magic we need.  

I'll move the check up a line now.

Reply:
Fixed already in my latest commit, unless you are suggesting something different?

Reply:
I see no commits, this has been merged to the repo.  See the purple 'Merged' button here.  

Note that I also just committed so if you want to change relative to the repo you need to update your for by pulling the main repo here.

Reply:
Your change seems perfectly fine - I often forget how GitHub works past midnight... Glad this was caught and fixed.

Reply:
Ugh - I spoke too soon. See my PR #106.